test/library/port_forwarder.go

```diff
@@ -0,0 +1,87 @@
+package library
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/portforward"
+	"k8s.io/client-go/transport/spdy"
+)
+
+// ForwardPodPortAndExecute forwards a local port to a pod port and executes the supplied func
+func ForwardPodPortAndExecute(t *testing.T, kubeConfig *rest.Config, pod *corev1.Pod, podPort int, consumer func(localPort int) error) error {
```

Comment:
this might be useful to have in operator-sdk, Have you thought of moving some functions there? I know some of them are quite specific to our use-case, but maybe some more general cases such as this one.